atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -211,8 +211,13 @@ private void testDataSource(HikariDataSource dataSourcePool) throws SQLException
      * @throws SQLException if connection is invalid and cannot be used.
      */
     private void testConnection(Connection connection) throws SQLException {
-        try (Statement stmt = connection.createStatement();
-                ResultSet rs = stmt.executeQuery(connConfig.getTestQuery())) {
+        testConnection(connection, Optional.empty());
```

Comment:
confused - what's the point of splitting out profiling into two steps if the profiler is never actually called because you're passing in Optional.empty()? seems like you need to pipe this one level higher.

Reply:
Testing the connection is done in two places - `acquirePooledConnection` and `testDataSource`, but we only have been doing the profiling in the first case. In the second case we don't have a profiler object, which is why we need a version that takes `Optional`. 

That said, I forgot to update the call inside `acquirePooledConnection` to use the profiler, which is now done - good catch there.